shell.nix

```diff
@@ -5,8 +5,24 @@ in
 pkgs.mkShell {
   name = "scripts-shell";
   buildInputs = with pkgs; [
+    chart-testing
+    ginkgo
+    git
+    go_1_19
+    golint
+    kubectl
+    kubernetes-helm
+    gnumake
+    minikube
     semver-tool
     yq-go
-    chart-testing
   ];
-}
\ No newline at end of file
+  shellHook = ''
+    export GOPATH=$(pwd)/nix/.go
+    export GOCACHE=$(pwd)/nix/.go/cache
+    export TMPDIR=$(pwd)/nix/.tmp
+    export PATH=$GOPATH/bin:$PATH
+    mkdir -p "$TMPDIR"
+    go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.0
```

Comment:
I think this is now the same the Makefile is installing? Do we need to do it here?

Reply:
But setting env variables was required, so I can run `make bootstrap` instead of `go install`

Reply:
Yeah that's fine, just wondering about specifically the install to avoid having to modify this file once we update the Makefile

Reply:
That said we also have the specific go version here, so we might need to change anyway :)

Reply:
Yeah, Also if we upgrade nixpkgs specified go version will not be available

Reply:
I would recommend changing the entire CI system to use nix-shell just like lint action

Reply:
Switched to `make bootstrap`, but getting

```
go: github.com/axw/gocov/gocov@latest (in github.com/axw/gocov@v1.2.1): go.mod:3: invalid go version '1.22.0': must match format 1.23
go.mod:5: unknown directive: toolchain
```

in the logs

but tools are installed correctly as far as I can tell

Reply:
Also getting

```
  go: gopkg.in/matm/v1/gocov-html@latest: unrecognized import path "gopkg.in": parse https://gopkg.in/?go-get=1: no go-import meta tags ()
```